File: javascript/src/domain/geometry/circle.js
module.exports = class Circle {

    constructor(name, radius) {
        this.name = name;
        this.radius = radius;
    }

    calculateArea() {
        return Math.PI * this.radius * this.radius;
    }

    calculateCircumference() {
        return 2 * Math.PI * this.radius;
    }

    getName() {
        return this.name;
    }
}

File: java/src/main/java/nl/craftsmen/workshops/reactivemeetup/domain/geometry/Circle.java
package nl.craftsmen.workshops.reactivemeetup.domain.geometry;

public class Circle implements Shape {

    private final String name;
    private final int radius;

    public Circle(String name, int radius) {
        this.name = name;
        this.radius = radius;
    }

    @Override
    public double calculateArea() {
        return Math.PI * radius * radius;
    }

    @Override
    public double calculateCircumference() {
        return 2 * Math.PI * radius;
    }

    @Override
    public String getName() {
        return name;
    }

}
